### ISSUE TITLE
Cross-reference links that are *not* header numbers

### ISSUE THREAD

**Issue by RexJaeschke** [spdx/spdx-spec#237]
Consider the following in file 3-package-information.md:

`… denoted by the [Package Name](#3.1) field.`

In md, this is rendered as 

`… denoted by the Package Name field.`

where “Package Name” is the hyperlink, which when selected leads to “3.3.1”. And that is the normal behavior/expectation in the md world. 

While the ISO Directives make _no_ mention of this approach, _all_ cross-reference instances in those Directives contain an _explicit_ link separate from the term to which the hyperlink pertains. As such, we deduce that the text above must be written instead, as follows:

`… denoted by the Package Name (see 3.3.1) field.`

with the clause/subclause number (rather than the text “Package Name”) being the hyperlink.

A similar situation exists with a hyperlink to an external document. For example:

Consider the following text (taken from 2.1 SPDX Version): 

`This document uses the prefix rdf: to refer to the [RDF/XML][rdf] namespace:`

where “RDF/XML” is a hyperlink, which when selected leads to the (otherwise hidden) URI
 https://www.w3.org/TR/2014/REC-rdf-syntax-grammar-20140225/”.

So, the text above must be written instead, as follows:

`This document uses the prefix rdf: to refer to the RDF/XML namespace (see https://www.w3.org/...-20140225/):`

with the explicit URI being the hyperlink.

Bottom line, each hyperlink **must show in printable form** the destination of the link (presumably so people using a hard-copy of the spec know where to look).

**Comment by RexJaeschke** [comment]
@kestewart: Consider the following:

"... to refer to the [RDF/XML][rdf] namespace:"

[rdf]: https://www.w3.org/TR/2014/REC-rdf-syntax-grammar-20140225/

which is written in md as 

`... to refer to the [RDF/XML][rdf] namespace:`

where 

`[rdf]: https://www.w3.org/TR/2014/REC-rdf-syntax-grammar-20140225/`

The corresponding ISO approach is to have the following link instead:

"... to refer to the RDF/XML namespace (see https://www.w3.org/TR/2014/REC-rdf-syntax-grammar-20140225/):"

ISO wants each hyperlink to show in printable form the destination of the link (presumably so people using a hard-copy of the spec know where to look).

Other examples are:

`[Uniform Resource Identifier][URI]` 

vs.

`Uniform Resource Identifier (see https://en.wikipedia.org/wiki/Uniform_Resource_Identifier)`

and

`[RFC-3986][rfc3986]`

vs.

`RFC3986 (see https://tools.ietf.org/html/rfc3986)`

We have two choices here:

1. We change all such links to the verbose format showing the explicit URL.
1. But if your audience prefers the current approach, since you are going to maintain the spec in md format, leave it as is and do this conversion for each future ISO submission.

Note that Approach 1 removes all the shortcuts such as `[rdf]` and `[uri]` requiring that their full expansions be hard-coded everywhere.

Which do you prefer?

**Comment by kestewart** [comment]
Lets hold off on unrolling these, until ISO makes an explicit request.
